lib/rubocop/cop/style/block_delimiters.rb

```diff
@@ -165,7 +164,7 @@ module Style
       #     process(something)
       #   }
       #
-      class BlockDelimiters < Base
+      class BlockDelimiters < Base # rubocop:disable Metrics/ClassLength
```

Comment:
Based on the rationale for the cop here, shouldn't this comment not be present and an exception added to `.rubocop.yml` instead? If this cop is a good idea for users, then it should be for RuboCop itself too.

Reply:
The purpose of this cop is not to recommend against disabling cops in source code entirely (that's covered by [`Style/DisableCopsWithinSourceCodeDirective`](https://docs.rubocop.org/rubocop/cops_style.html#styledisablecopswithinsourcecodedirective)). It is only meant to flag the use of `rubocop:disable` around the entire file.

In this case, addressing the cop could happen in three ways:
- Move the config to `.rubocop.yml`
  - This is okay, but it means the cop is effectively still disabled across the entire file
- Move the `rubocop:disable`/`rubocop:enable` to just be around the offending class
  - This is better, but it means the cop is still disabled for any nested classes that might be defined
- Target this specific class when disabling, so the cop still applies to any inner classes
  - This is ideal, because it conveys exactly what we want: "We're okay with the length of the `BlockDelimiters`."

Reply:
I understand, and I'm not necessarily suggesting that its detection should change at all. But since almost every ruby file contains a single class, the following are practically the same:

```ruby
# rubocop:disable Foo/Bar
class X
  # ...
end
# rubocop:enable Foo/Bar
```

```ruby
class X
# rubocop:disable Foo/Bar
  # ...
# rubocop:enable Foo/Bar
end
```

The latter won't register the cop though obviously. So in the instances where you have replaced the former with the latter, IMO it'd be better to just put the disable into .rubocop.yml because otherwise it's just shuffling code to satisfy a cop. Does that make sense?

Reply:
I agree given your example, but note that the code actual code switches to an inline directive, not a block one,

```diff
-# rubocop:disable Foo/Bar
-class X
+class X # rubocop:disable Foo/Bar
   # ...
 end
-# rubocop:enable Foo/Bar
```

which applies to only **that** class and **not to any nested classes**,

```ruby
class ExceptionallyLongClass # rubocop:disable Metrics/ClassLength
  class MustBeShortClass
  end
end
```

whereas the block comment or config comment would apply to the entire range or file.

```ruby
# rubocop:disable Metrics/ClassLength
class ExceptionallyLongClass
  class CanAlsoBeLongClass
  end
end
# rubocop:enable Metrics/ClassLength
```